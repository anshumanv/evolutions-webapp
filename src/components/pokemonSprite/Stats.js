import React from 'react';
import PropTypes from 'prop-types';

const Stats = ({ baseStats }) => (
  <React.Fragment>
    <div> Stats: </div>
    <div className="PokemonSprite-row">
      <div className="PokemonSprite-col">Hp: {baseStats.hp} </div>
      <div className="PokemonSprite-col">Atk: {baseStats.atk} </div>
      <div className="PokemonSprite-col">Def: {baseStats.def} </div>
    </div>
    <div className="PokemonSprite-row">
      <div className="PokemonSprite-col"> Sp.Atk: {baseStats.spa} </div>
      <div className="PokemonSprite-col"> Sp.Def: {baseStats.def} </div>
      <div className="PokemonSprite-col"> Sp: {baseStats.spe} </div>
    </div>
  </React.Fragment>
);

Stats.propTypes = {
  baseStats: PropTypes.object,
};

export default Stats;
