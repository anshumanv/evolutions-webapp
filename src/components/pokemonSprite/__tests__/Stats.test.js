import React from 'react';
import { shallow } from 'enzyme';
import Stats from '../Stats';

describe('Stats component', () => {
  it('renders', () => {
    const sampleData = {
      baseStats: {
        hp: 35,
        atk: 55,
        def: 40,
        spa: 50,
        spd: 40,
        sp: 90,
      },
    };
    const component = shallow(<Stats baseStats={sampleData} />);
    expect(component).toMatchSnapshot();
  });
});
