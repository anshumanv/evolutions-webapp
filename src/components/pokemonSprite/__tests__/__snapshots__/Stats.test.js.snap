// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Stats
    baseStats={
        Object {
            "baseStats": Object {
              "atk": 55,
              "def": 40,
              "hp": 35,
              "sp": 90,
              "spa": 50,
              "spd": 40,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
           Stats: 
</div>,
        <div
          className="PokemonSprite-row"
>
          <div
                    className="PokemonSprite-col"
          >
                    Hp: 
                     
          </div>
          <div
                    className="PokemonSprite-col"
          >
                    Atk: 
                     
          </div>
          <div
                    className="PokemonSprite-col"
          >
                    Def: 
                     
          </div>
</div>,
        <div
          className="PokemonSprite-row"
>
          <div
                    className="PokemonSprite-col"
          >
                     Sp.Atk: 
                     
          </div>
          <div
                    className="PokemonSprite-col"
          >
                     Sp.Def: 
                     
          </div>
          <div
                    className="PokemonSprite-col"
          >
                     Sp: 
                     
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Stats: ",
        },
        "ref": null,
        "rendered": " Stats: ",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="PokemonSprite-col"
>
              Hp: 
               
</div>,
            <div
              className="PokemonSprite-col"
>
              Atk: 
               
</div>,
            <div
              className="PokemonSprite-col"
>
              Def: 
               
</div>,
          ],
          "className": "PokemonSprite-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Hp: ",
                undefined,
                " ",
              ],
              "className": "PokemonSprite-col",
            },
            "ref": null,
            "rendered": Array [
              "Hp: ",
              undefined,
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Atk: ",
                undefined,
                " ",
              ],
              "className": "PokemonSprite-col",
            },
            "ref": null,
            "rendered": Array [
              "Atk: ",
              undefined,
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Def: ",
                undefined,
                " ",
              ],
              "className": "PokemonSprite-col",
            },
            "ref": null,
            "rendered": Array [
              "Def: ",
              undefined,
              " ",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="PokemonSprite-col"
>
               Sp.Atk: 
               
</div>,
            <div
              className="PokemonSprite-col"
>
               Sp.Def: 
               
</div>,
            <div
              className="PokemonSprite-col"
>
               Sp: 
               
</div>,
          ],
          "className": "PokemonSprite-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Sp.Atk: ",
                undefined,
                " ",
              ],
              "className": "PokemonSprite-col",
            },
            "ref": null,
            "rendered": Array [
              " Sp.Atk: ",
              undefined,
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Sp.Def: ",
                undefined,
                " ",
              ],
              "className": "PokemonSprite-col",
            },
            "ref": null,
            "rendered": Array [
              " Sp.Def: ",
              undefined,
              " ",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " Sp: ",
                undefined,
                " ",
              ],
              "className": "PokemonSprite-col",
            },
            "ref": null,
            "rendered": Array [
              " Sp: ",
              undefined,
              " ",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
             Stats: 
</div>,
          <div
            className="PokemonSprite-row"
>
            <div
                        className="PokemonSprite-col"
            >
                        Hp: 
                         
            </div>
            <div
                        className="PokemonSprite-col"
            >
                        Atk: 
                         
            </div>
            <div
                        className="PokemonSprite-col"
            >
                        Def: 
                         
            </div>
</div>,
          <div
            className="PokemonSprite-row"
>
            <div
                        className="PokemonSprite-col"
            >
                         Sp.Atk: 
                         
            </div>
            <div
                        className="PokemonSprite-col"
            >
                         Sp.Def: 
                         
            </div>
            <div
                        className="PokemonSprite-col"
            >
                         Sp: 
                         
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Stats: ",
          },
          "ref": null,
          "rendered": " Stats: ",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="PokemonSprite-col"
>
                Hp: 
                 
</div>,
              <div
                className="PokemonSprite-col"
>
                Atk: 
                 
</div>,
              <div
                className="PokemonSprite-col"
>
                Def: 
                 
</div>,
            ],
            "className": "PokemonSprite-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Hp: ",
                  undefined,
                  " ",
                ],
                "className": "PokemonSprite-col",
              },
              "ref": null,
              "rendered": Array [
                "Hp: ",
                undefined,
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Atk: ",
                  undefined,
                  " ",
                ],
                "className": "PokemonSprite-col",
              },
              "ref": null,
              "rendered": Array [
                "Atk: ",
                undefined,
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Def: ",
                  undefined,
                  " ",
                ],
                "className": "PokemonSprite-col",
              },
              "ref": null,
              "rendered": Array [
                "Def: ",
                undefined,
                " ",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="PokemonSprite-col"
>
                 Sp.Atk: 
                 
</div>,
              <div
                className="PokemonSprite-col"
>
                 Sp.Def: 
                 
</div>,
              <div
                className="PokemonSprite-col"
>
                 Sp: 
                 
</div>,
            ],
            "className": "PokemonSprite-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Sp.Atk: ",
                  undefined,
                  " ",
                ],
                "className": "PokemonSprite-col",
              },
              "ref": null,
              "rendered": Array [
                " Sp.Atk: ",
                undefined,
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Sp.Def: ",
                  undefined,
                  " ",
                ],
                "className": "PokemonSprite-col",
              },
              "ref": null,
              "rendered": Array [
                " Sp.Def: ",
                undefined,
                " ",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " Sp: ",
                  undefined,
                  " ",
                ],
                "className": "PokemonSprite-col",
              },
              "ref": null,
              "rendered": Array [
                " Sp: ",
                undefined,
                " ",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
